utils/src/lib.rs

```diff
@@ -143,6 +143,26 @@
 )]
 // When we use rust version 1.65 or later, refactor this with GAT
 
+use std::str::FromStr;
+
+#[cfg(not(madsim))]
+use tonic::transport::ClientTlsConfig;
+use tonic::transport::Endpoint;
+
+/// Fake `ClientTlsConfig` for `madsim`
+#[cfg(madsim)]
+#[derive(Debug, Clone)]
+#[allow(missing_copy_implementations)] // Same as real config
+#[non_exhaustive]
+pub struct ClientTlsConfig;
+
+/// Fake `ServerTlsConfig` for `madsim`
+#[cfg(madsim)]
+#[derive(Debug, Clone)]
+#[allow(missing_copy_implementations)] // Same as real config
+#[non_exhaustive]
+pub struct ServerTlsConfig;
```

Comment:
Is it better to place these two struct in `simulation/src/lib.rs`?

Reply:
`xline` also depends on these fake configs, and `xline` shouldn't depend on `simulation`